05.wc/wc.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = define_options
+
+  if stand_alone?
+    files = ARGV
+    file_details = files.to_h { |file| [file, File.read(File.expand_path(file))] }
+    file_details_total = Hash.new(0)
+
+    file_details.each do |file_name, file_string|
+      detail_line = create_detail_line(options, file_string)
+      puts "#{output_lines(detail_line)} #{file_name}"
+
+      options.each do |option|
+        file_details_total[option] += detail_line[option]
+      end
+    end
+
+    output_total_line(file_details_total) if file_details.size > 1
+  else
+    detail_line = create_detail_line(options, $stdin.to_a.join)
+    puts output_lines(detail_line)
+  end
+end
+
+def create_detail_line(options, file_string)
```

Comment:
line = 行 = 文字列？みたいな印象を受けます。
ハッシュを返すなら

```suggestion
def create_details(options, file_string)
```

のように複数形にするだけでいいかも。

あとoptionsって、あくまでオプションなので、引数の先頭に来ることはあまりない気がしますね。

```suggestion
def create_details(file_string, options)
```